src/Stratis.Bitcoin.Features.PoA/FederationManager.cs

```diff
@@ -10,29 +10,50 @@
 
 namespace Stratis.Bitcoin.Features.PoA
 {
-    public class FederationManager
+    public interface IFederationManager
+    {
+        bool IsFederationMember { get; }
+
+        Key CurrentFederationKey { get; }
+
+        void Initialize();
+
+        /// <summary>Provides up to date list of federation members.</summary>
+        /// <remarks>
+        /// Blocks that are not signed with private keys that correspond
+        /// to public keys from this list are considered to be invalid.
+        /// </remarks>
+        List<IFederationMember> GetFederationMembers();
+
+        void AddFederationMember(IFederationMember federationMember);
+
+        void RemoveFederationMember(IFederationMember federationMember);
+    }
+
+    public class FederationManager : IFederationManager
     {
         /// <summary><c>true</c> in case current node is a federation member.</summary>
```

Comment:
Perhaps these comments should be moved to the interface definition and inherited?

Reply:
Fixed